pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+            self.accelerator = self._accelerator_flag
+        elif self._accelerator_flag == "tpu":
+            self.accelerator = TPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = TPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int):
+                self._parallel_devices = list(range(self._device_flag))
+            else:
+                self._parallel_devices = self._device_flag
+
+        elif self._accelerator_flag == "ipu":
+            self.accelerator = IPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = IPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int):
+                self._parallel_devices = list(range(self._device_flag))
+
+        elif self._accelerator_flag == "gpu":
+            self.accelerator = GPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = GPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int) or isinstance(self._device_flag, str):
+                self._device_flag = int(self._device_flag)
+                self._parallel_devices = [
+                    torch.device("cuda", i) for i in device_parser.parse_gpu_ids(self._device_flag)
+                ]
+            else:
+                self._parallel_devices = [torch.device("cuda", i) for i in self._device_flag]
+
+        elif self._accelerator_flag == "cpu":
+            self.accelerator = CPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = CPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int):
+                self._parallel_devices = [torch.device("cpu")] * self._device_flag
+            else:
+                rank_zero_warn(
                     "The flag `devices` must be an int with `accelerator='cpu'`,"
-                    f" got `devices={self.devices}` instead."
+                    f" got `devices={self._device_flag}` instead."
                 )
-            self.num_processes = self.devices
-            return True
-        return False
 
-    @property
-    def use_dp(self) -> bool:
-        return self._strategy_type == _StrategyType.DP
+        self._gpus = self._device_flag if not self._gpus else self._gpus
 
-    @property
-    def use_ddp(self) -> bool:
-        return self._strategy_type in (
-            _StrategyType.DDP,
-            _StrategyType.DDP_SPAWN,
-            _StrategyType.DDP_SHARDED,
-            _StrategyType.DDP_SHARDED_SPAWN,
-            _StrategyType.DDP_FULLY_SHARDED,
-            _StrategyType.DEEPSPEED,
-            _StrategyType.TPU_SPAWN,
-        )
+    def _choose_and_init_cluster_environment(self):
+        self.cluster_environment = LightningEnvironment()
```

Comment:
why do we set the cluster environment here first? it's confusing to read the code and see it overridden later
why do we have both self._cluster_environment and self.cluster_environment?